common-libs/user-management/src/CognitoIdentityService.ts

```diff
@@ -196,6 +198,24 @@ export class CognitoIdentityService {
       const cognitoTokens = this._normalizeTokensFromCognitoAuthenticationResult(result.AuthenticationResult)
       return { result: CompleteLoginPasswordlessResult.Success, cognitoTokens }
     } catch (error) {
+      // NOTE: IF below is a fix: when user first time enters wrong OTP, Cognito
+      //       returns new session, for which correct OTP wont work, so user
+      //       will be stuck until server restart and user enters OTP correctly
+      //       from the 1st attempt
+      if (error?.code === 'COR-5' && error?.inputParams?.token) {
```

Comment:
COR-5 is our error code, as far as I can see there is no way it can be thrown from this `try` block, so this is effectively a no-op?

Also could you please write some integration tests covering this scenario? (They don't have to be in this library, can be in `sdk/core/test/integration/otp` as well)

Reply:
It seems that you need to add your logic to `if (result.ChallengeName === 'CUSTOM_CHALLENGE') {` on line 194 instead of that `catch` block.